team51-tracking.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * The Team51 Tracking plugin bootstrap file.
+ *
+ * @since       1.0.0
+ * @version     1.0.0
+ * @author      WordPress.com Special Projects
+ * @license     GPL-3.0-or-later
+ *
+ * @noinspection    ALL
+ *
+ * @wordpress-plugin
+ * Plugin Name:             Team51 Tracking
+ * Description:             Handles automatic opt-in on parter sites to usage tracking for WooCommerce, Sensei, and other plugins/tools used by Team51.
+ * Version:                 1.0.0
+ * Requires at least:       6.1
+ * Requires PHP:            8.1
+ * Author:                  WordPress.com Special Projects
+ * Author URI:              https://wpspecialprojects.wordpress.com/
+ * License:                 GPL-3.0-or-later
+ * License URI:             https://www.gnu.org/licenses/gpl-3.0.html
+ * Text Domain:             team51-tracking
+ * Domain Path:             /languages
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+// Define plugin constants.
+function_exists( 'get_plugin_data' ) || require_once ABSPATH . 'wp-admin/includes/plugin.php';
+define( 'WPCOMSP_TRACKING_METADATA', get_plugin_data( __FILE__, false, false ) );
+
+define( 'WPCOMSP_TRACKING_BASENAME', plugin_basename( __FILE__ ) );
+define( 'WPCOMSP_TRACKING_PATH', plugin_dir_path( __FILE__ ) );
+define( 'WPCOMSP_TRACKING_URL', plugin_dir_url( __FILE__ ) );
+
+// Load plugin translations so they are available even for the error admin notices.
+add_action(
+	'init',
+	static function() {
```

Comment:
I'm curious what's the use of `static` here in this anonymous function.

Reply:
It's a small performance optimization 😄 When declared non-static, the parser will bind the function to the outer scope and have it inherit things like variables. When it's static, that step is skipped and thus it's slightly faster. Not a big deal, but it just feels "correct" to me ... and I also get a hint from one of my PhpStorm extensions about this whenever declaring a function static makes sense.